### ISSUE TITLE
Problem with 4-digit octal values generated in file paths from zfs diff

### ISSUE THREAD

**Issue by schwerpunkt** [schwerpunkt/zfsDiffReport.py#6]
Breaks getHash function because special characters such as ü = \\0303\\0274 or space = \\0040 are not decoded properly. (actual problem lies in [zfs implementation](https://github.com/zfsonlinux/zfs/issues/6318#issuecomment-313355603))
Implement workaround / proper decoding in python after diff list is read from zfs out in *getSortedDiffLines()*.

**Comment by schwerpunkt** [comment]
https://github.com/schwerpunkt/zfsDiffReport.py/blob/4a008a8ec0ffc902651ed09c07bbbdcf00d5d646/zfsDiffReport.py#L82-L89

Unacceptable.

**Comment by hungrywolf27** [comment]
I have a similar script to parse zfs diff, and this is the solution I came up with through a lot of trial and error:

```python
import re

def decode_octal(encoded):
    # Match b'\\0xxx'
    for oct_char in (c for c in re.findall(b'\\\\0\d{3}', encoded)):
        # Figured out through hours of trial and error:
        # Strip out '\0' to leave b'xxx' of octal number
        #    br'' means raw bytes (\ not treated as escape)
        #    Note that it's impossible to create b'\' in Python;
        #    can't end with a single backslash.
        myoct = oct_char.replace(br'\0', b'')
        # Convert this from octal to an integer
        myint = int(myoct, 8)
        encoded = encoded.replace(oct_char, bytes([myint]))
    return encoded
```

**Comment by hungrywolf27** [comment]
For example:
```python
a = decode_octal(b'H\\0303\\0251ll\\0303\\0266\\0040world')
print(a.decode('utf-8'))
```

**Comment by schwerpunkt** [comment]
Thanks a lot for your comment. I included the code (#13) replacing my non-general character replace routine. I remember the pain researching a solution only to eventually table-flip-quitting and going for the quick fix, so thanks again for sharing!
